### ISSUE TITLE
Add API to get the selector for an Exported method

### ISSUE THREAD

**Issue by charabaruk** [dotnet/macios#6278]
_Submitted by a community member on 2012-07-02 18:33 UTC_

I am trying to implement a proxy pattern using the ObjectiveC mechanisms for transparent forwarding of messages. Basically I want an object to respond to a selector even though there is no real method with that selector. There are three steps to this:
1\. Override respondsToSeletor: to return true for the selector you are faking.
2\. Override methodSignatureForSelector: to return a method signature for the selector which does not exist. I do this by returning the signature for a compatible method which does exist.
3\. Override forwardInvocation: to intercept the actual call to the given method and make it do something entirely different.

When I attempt step #‌2 in MonoTouch the runtime crashes as soon as I create an instance of my object. The relevant code that causes the crash (though does not appear to actually be executed) is this:

```
/// BEGIN CODE
private const string MethodSignatureForSelectorSelectorName = "methodSignatureForSelector:";
private static readonly Selector methodSignatureForSelectorSelector = new Selector(MethodSignatureForSelectorSelectorName);

[Export(MethodSignatureForSelectorSelectorName)]
IntPtr MethodSignatureForSelector(Selector sel)
{
    if (sel.Name == DummyActionSelectorName)
    {
        sel = PrototypeMethodSelector;
    }
    
    return Messaging.IntPtr_objc_msgSend_IntPtr(SuperHandle, methodSignatureForSelectorSelector.Handle, sel.Handle);

}
/// END CODE
```

It appears that the mere existence of this code in a class causes allocating that class to crash the runtime. If I set a breakpoint in that method it never gets hit.

Attached is a .zip containing two sample projects: one written in MonoTouch which demonstrates the crash, and the other written in ObjectiveC which demonstrates the intended behavior. To reproduce the crash just run the MonoTouch application and press the button labeled "Crashes".

* * *

_Reference: <https://bugzilla.xamarin.com/show_bug.cgi?id=5954>_

**Comment by coldacid** [comment]
Comment 8 from the original issue, by @rolfbjarne, would be a reasonable solution to this problem and would still be of use for anyone who needs to work with UIMenuController for custom and dynamic edit menu items.

**Comment by rolfbjarne** [comment]
@coldacid is your use case the same as the original reporter's (and would his workaround work for you as well)?

**Comment by coldacid** [comment]
@rolfbjarne yeah, use case is the same, and for now I'm taking the same workaround approach.
